File: LeetCode Solutions/Binary Search II/Frequency of the Most Frequent Element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        ans = 1
        nums=sorted(nums)
        sums = 0
        l = 0
        for r in range(len(nums)):
            sums += nums[r]
            while l < r and sums + k <  nums[r] * (r - l + 1):
                sums -= nums[l]
                l += 1
            ans = max(ans, r - l + 1)
        return ans

File: LeetCode Solutions/Study Plan/Binary Search Study Plan/Binary Search II/Frequency of the Most Frequent Element.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
        int result = 1, l = 0, r;
        long sum = 0;
        Arrays.sort(nums);
        for (r=0; r<nums.length; ++r) {
            sum += nums[r];
            while (sum + k < (long)nums[r] * (r - l + 1)) {
                sum -= nums[l];
                l += 1;
            }
            result = Math.max(result, r - l + 1);
        }
        return result;
    }
}